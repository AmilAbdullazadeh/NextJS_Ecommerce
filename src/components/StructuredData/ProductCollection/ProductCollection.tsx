import React from 'react';

import { SITE_URL } from '@core';

export interface ProductCollectionProps {
    size: number;
    headline: string;
    keywords: string;
    description: string;
}

const ProductCollection: React.FC<ProductCollectionProps> = ({
    size,
    headline,
    keywords,
    description,
}: ProductCollectionProps) => {
    const JSON_LD = {
        '@context': 'http://schema.org',
        '@type': 'ProductCollection',
        size,
        headline,
        keywords,
        mainEntityOfPage: SITE_URL,
        url: SITE_URL,
        description,
    };

    return (
        <script
            type="application/ld+json"
            dangerouslySetInnerHTML={{ __html: JSON.stringify(JSON_LD) }}
        ></script>
    );
};

export default ProductCollection;
