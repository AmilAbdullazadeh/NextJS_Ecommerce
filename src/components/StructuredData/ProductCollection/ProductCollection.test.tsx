import React from 'react';

import { render } from '@testUtils';

import ProductCollection, { ProductCollectionProps } from './ProductCollection';

describe('ProductCollection', () => {
    const defaultProps: ProductCollectionProps = {
        size: 10,
        headline: 'mock headline',
        keywords: 'keyword-1,keyword-2',
        description: 'mock description',
    };

    it('renders correctly', () => {
        const {
            container: { firstChild },
        } = render(<ProductCollection {...defaultProps} />);

        expect(firstChild).toMatchSnapshot();
    });
});
