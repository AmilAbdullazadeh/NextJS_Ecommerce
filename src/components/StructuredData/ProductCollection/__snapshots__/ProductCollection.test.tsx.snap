// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCollection renders correctly 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"http://schema.org","@type":"ProductCollection","size":10,"headline":"mock headline","keywords":"keyword-1,keyword-2","mainEntityOfPage":"http://localhost:3000","url":"http://localhost:3000","description":"mock description"}
</script>
`;
